main.tf

```diff
@@ -200,3 +204,28 @@ resource "aws_elasticache_replication_group" "this" {
     aws_security_group.this
   ]
 }
+
+resource "aws_elasticache_serverless_cache" "serverless" {
+  count = var.cache_type == "serverless" ? 1 : 0
+
+  engine             = var.engine
+  name               = var.name
+  security_group_ids = var.create_security_group ? [aws_security_group.this[0].id] : var.security_group_ids
+  subnet_ids         = var.subnet_ids
+
+  cache_usage_limits {
+    data_storage {
+      maximum = var.data_storage_maximum
+      unit    = "GB"
+    }
+    ecpu_per_second {
+      maximum = var.ecpu_per_second_maximum
+    }
+  }
+
+  tags = var.serverless_tags
+
+  depends_on = [
+    aws_security_group.this
+  ]
+}
```

Comment:
* If you are going to add a new resource, it **should match any use case with the full configuration**. **[Reference](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/elasticache_serverless_cache)**.